core/src/serve_repair.rs

```diff
@@ -258,6 +258,31 @@ pub enum RepairProtocol {
     },
 }
 
+const REPAIR_REQUEST_PONG_SERIALIZED_BYTES: usize = PUBKEY_BYTES + HASH_BYTES + SIGNATURE_BYTES;
+const REPAIR_REQUEST_MIN_BYTES: usize = REPAIR_REQUEST_PONG_SERIALIZED_BYTES;
+
+fn check_well_formed_repair_request(packet: &Packet, stats: &mut ServeRepairStats) -> Result<()> {
+    if packet.meta().size < REPAIR_REQUEST_MIN_BYTES {
+        stats.err_malformed += 1;
+        return Err(Error::from(RepairVerifyError::Malformed));
+    }
+    Ok(())
+}
```

Comment:
Seems overkill to have a function just to check `packet.meta().size < REPAIR_REQUEST_MIN_BYTES`.
Also the `Error` value is immediately discarded.
Can we just check `packet.meta().size` inside `discard_malformed_repair_requests`